src/Classes/ServiceAPI/MyRadio_Scheduler.php

```diff
@@ -187,7 +187,8 @@ public static function getScheduleConflicts($term_id, $time)
         $conflicts = [];
         $start_day = (new MyRadio_Term($term_id))->getTermStartDate() + ($time['day'] * 86400);
         //Iterate over each week
-        for ($i = 1; $i <= 10; ++$i) {
+        //Changes to 11 weeks for semesters
+        for ($i = 1; $i <= 11; ++$i) {
```

Comment:
So this works, but there's no guarantee that semesters will always be 11 weeks - is there a way that we can avoid hard-coding a number here completely?

Reply:
I changed it to use the num_weeks from Myradio_term. 
Is this okay or do I need to get the number weeks from somewhere else

Reply:
You're on the right track. As written it won't work, because`getTermWeeks` is a method on `MyRadio_Term` so you need to call it on a term object (`$term->getTermWeeks()`). You'd need to do something like `(new MyRadio_Term($term_id))->getTermWeeks()`.

However, since there's already another MyRadio_Term constructor above, you could remove the duplication:

```php
$term = new MyRadio_Term($term_id);
$start_day = $term->getTermStartDate() + ($time['day'] * 86400);

for ($i = 1; $i <= $term->getTermWeeks(); ++$i) {
// ...
}
```